test/unit_tests/arkode/C_serial/ark_test_adapt.c

```diff
@@ -0,0 +1,154 @@
+/* -----------------------------------------------------------------------------
+ * Programmer(s): Steven B. Roberts @ LLNL
+ * -----------------------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------------------
+ * Unit test for step size growth and handling of inf/nan within a controller
+ * due to 0 local error.
+ * ---------------------------------------------------------------------------*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "arkode/arkode_erkstep.h"
+#include "nvector/nvector_serial.h"
+
+/* If an error occurs, print to stderr and exit */
+static void err_fn(const int line, const char* const func, const char* const file,
+                   const char* const msg, const SUNErrCode err_code,
+                   void* const err_user_data, const SUNContext sunctx)
+{
+  fprintf(stderr, "Error at line %i of %s in %s: %s\n", line, func, file, msg);
+  exit(err_code);
+}
+
+/* RHS for the simple ODE y'=0 */
+static int f(const sunrealtype t, const N_Vector y, const N_Vector ydot,
+             void* const user_data)
+{
+  N_VConst(SUN_RCONST(0.0), ydot);
+  return 0;
+}
+
+/* Take a single time step solving y'=0 and check the max growth was attained */
+static int check_step(void* const arkode_mem, const N_Vector y,
+                      const sunrealtype h_expected, const int step)
+{
+  /* Integration much farther than expected step */
+  const sunrealtype tout = SUN_RCONST(100.0) * h_expected;
+  sunrealtype tret;
+  /* The ERK method should be able to take the maximum possible timestep for the
```

Comment:
```suggestion
  /* The ERK method should be able to take the maximum possible timestep
```

Reply:
This was code merged from https://github.com/LLNL/sundials/pull/548, so I'll make the fix there and merge it into this branch.